New feature: RSS

imba3r opened the pull request:
Hi,

great library - thank you for this! I needed RSS capabilities, maybe you are interested in this. Most indexers that I'm aware of have a RSS feed under /rss whereas the normal api is under /api. The results and query parameters are pretty much the same. Note that users will have to specify their indexer base URL without the trailing /api in it - so this is kind of a breaking change. Let me know if you are interested and I will also update the Readme.

Best regards, 
imba3r


* Build RSS / API url based on called function.
* Fetch RSS results (possible to specify count).
* Fetch RSS up to a given NZB id.
* Parse usenetdate (different to pubdate).

mrobinsn commented:
@imba3r This looks great! 

Please go ahead and update the `README` with the correct base URL. Then I will see no reason not to go ahead and merge this in. I will create a new major release noting the breaking change -- shouldn't be a problem.

imba3r commented:
Hi @mrobinsn,

I updated the readme about the /api and /rss base URL suffix. I also added another commit that adds two minor things:

- Attempt to parse date in RFC3339 if RFC1123Z failed (I've seen those a couple of times).
- Added Error Code + Description to Response struct, if those are filled return an error with the description.

If you don't want that in this Pull-Request I can leave that out.

Best regards,
imba3r
